packages/components/src/components/my-position/my-position.tsx

```diff
@@ -0,0 +1,402 @@
+                    accurate: this.positionIsAccurate
+                });
+            },
+            error => {
+                // Firefox may throw both success and a timeout error (https://bugzilla.mozilla.org/show_bug.cgi?id=1283563).
+                // We mitigate this by not passing on error if a correct position has been given since asking for it.
+                if (error.code !== 3 || !this.currentPosition || this.currentPosition.timestamp <= requestTime) {
+                    this.setPositionUnknown(error);
+                }
+            },
+            this.options.positionOptions
+        );
+    }
+    /**
+     * Sets position button state.
+     *
+     * @param {PositionStateTypes} state
+     */
+    private setPositionState(state: PositionStateTypes): void {
+        this.positionState = state;
+    }
+
+    /**
+     * Handle click on the compass button.
+     */
+    private onCompassButtonClicked(): void {
+        this.setCompassStyle(0);
+        this.mapView.rotate(0);
+        this.mapView.tilt(0);
+        window.removeEventListener('deviceorientation', this.handleDeviceOrientationReference);
+
+        if (this.positionState === PositionStateTypes.POSITION_TRACKED) {
+            this.setPositionState(PositionStateTypes.POSITION_CENTERED);
+        } else if (this.positionState === PositionStateTypes.POSITION_UNTRACKED) {
+            this.setPositionState(PositionStateTypes.POSITION_KNOWN);
+        }
+    }
+
+    /**
+     * Handle click on position button.
+     * Stops propagation to avoid any map click listeners to fire.
+     *
+     * @param {Event} event
+     */
+    private onPositionButtonClicked(event: Event): void {
+        event.stopPropagation();
+        switch (this.positionState) {
+            case PositionStateTypes.POSITION_UNKNOWN:
+                this.mapView.tilt(0);
+                this.watchPosition(true);
+                break;
+            case PositionStateTypes.POSITION_KNOWN:
+                this.mapView.tilt(0);
+                this.panToCurrentPosition();
+                break;
+            case PositionStateTypes.POSITION_CENTERED:
+                if (this.canBeTracked) {
+                    this.setPositionState(PositionStateTypes.POSITION_TRACKED);
+                }
+                this.panToCurrentPosition();
+                break;
+            case PositionStateTypes.POSITION_TRACKED:
+                this.mapView.tilt(0);
+                this.mapView.rotate(0);
+                this.setCompassStyle(0);
+                this.setPositionState(PositionStateTypes.POSITION_CENTERED);
+                this.panToCurrentPosition();
+                break;
+            case PositionStateTypes.POSITION_UNTRACKED:
+                this.setPositionState(PositionStateTypes.POSITION_TRACKED);
+                this.panToCurrentPosition();
+                break;
+        }
+    }
+
+    /**
+     * Called every time the component is connected to the DOM.
```

Comment:
And what do we do in this callback? 😄 Lots of stuff happening, and it would be nice with a short summary.

Reply:
🙌 !